### PR TITLE
Fix statistics wedge not showing correct values in daily challenge

### LINKED ISSUE #29094: Statistics wedge shows values of 0 in daily challenge
### Type

Cosmetic

### Bug description

Statistics wedge shows values of 0 in daily challenge.

### Screenshots or videos

image

### Version

2024.726.0

### Logs

.

### PR DESCRIPTION
**Pull request by cbrwn**
Hey friends! Thought this was a decent issue to look at as my first contribution 😃 

Fixes #29094

I did wonder if this should be done in a value changed binding on `OsuScreen.Beatmap`, but this seems simple & clear enough that this should happen when we do other beatmap setting stuff

### CODE REVIEW

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs [hunk 1]
```diff
@@ -379,6 +379,9 @@ private void trySetDailyChallengeBeatmap()
             var beatmap = beatmapManager.QueryBeatmap(b => b.OnlineID == playlistItem.Beatmap.OnlineID);
             Beatmap.Value = beatmapManager.GetWorkingBeatmap(beatmap); // this will gracefully fall back to dummy beatmap if missing locally.
             Ruleset.Value = rulesets.GetRuleset(playlistItem.RulesetID);
+
+            userModsSelectOverlay.Beatmap.Value = Beatmap.Value;
```
[the remark this answers is not in the document]
Comment (cbrwn) [reply osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs:383]: aaaaaaaaaaah `BindTarget` - I initially tried something like that but couldn't figure it out! That's way nicer and totally just works, thanks for that!

**Comment by cbrwn** [comment]
Turns out I'm a noob at GitHub's pull request flow - I think I totally could've taken your change directly from that suggestion @bdach 😅

**Comment by bdach** [comment]
> I think I totally could've taken your change directly from that suggestion

Nah that was not a github suggestion, that was a git patch. If you could commit directly via github there'd be a button (but there's none).

**Comment by cbrwn** [comment]
> Nah that was not a github suggestion, that was a git patch. If you could commit directly via github there'd be a button (but there's none).

Ah gotcha! Is there anything else I needa do for this?

**Review by bdach — APPROVED** [review]

**Comment by bdach** [comment]
> Ah gotcha! Is there anything else I needa do for this?

Nope

**Pull request merged**